esptfa_arima/arima_model/arima_model.py

```
@@ -95,140 +189,228 @@ def make_stationary(student_data):
+
+    # drop the student_id from the features_df since it is just an identifier
+    features_df = features_df.drop("student_id", axis=1)
+
+    # transform into numpy array
+    features_df = features_df.to_numpy()
 
+    # get the model by pickling
+    model_path = os.path.join(os.path.dirname(__file__), "models", "esptfa_xgboost.pkl")
+    if not os.path.exists(model_path):
+        raise FileNotFoundError("Model not found at path: {}".format(model_path))
+    model = pickle.load(open(model_path, "rb"))
```

Ruff (0.14.10)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)